libraries/AP_OpticalFlow/AP_OpticalFlow_PX4Flow.cpp

```diff
@@ -83,7 +83,7 @@ bool AP_OpticalFlow_PX4Flow::scan_buses(void)
         }
         retry_attempt++;
         if (!success) {
-            hal.scheduler->delay(10);
+            hal.scheduler->delay(1000);
```

Comment:
One second of delay per retry seems way too much to me. What other values have you tried?

Reply:
PX4Flow need about 5 second to bootup, how about add a 5 second delay before attempt to try?

Reply:
I think that at best it should be `5 - system_time`. I would like to bring @tridge into this though, I think he was the one introducing the retries.

Reply:
the flow should already have had 5s from the start as the bootloader for ArduPilot waits for a USB connection for 5s by default.
It is possible the recent changes to disable and re-enable peripheral power while booting (at a suggestion from @proficnc) has triggered the issue?
One possible solution for sensors like these is to bring up the driver in an unhealthy state, then mark it healthy once it comes alive.
The reason I don't like long sleeps in drivers is that if several drivers ask for this it could take a very long time to boot.

Reply:
@tridge It feels like bootloader wait time was 0s in past few weeks ago. But I checked again, It looks normal now. If the 5s bootloader time is still there, then let's close this.